<launch>
  

  <arg name="odom_frame_id"   default="odom"/>
  <arg name="base_frame_id"   default="chassis"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="odom_topic" default="odom" />
  <arg name="laser_topic" default="scan" />


  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find navros_pkg)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find navros_pkg)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />   
    <rosparam file="$(find navros_pkg)/param/local_costmap_params.yaml" command="load" />   
    <rosparam file="$(find navros_pkg)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find navros_pkg)/param/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find navros_pkg)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find navros_pkg)/param/navfn_global_planner_params.yaml" command="load" />
    
    
    <!-- reset frame_id parameters using user input data -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>

    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
  </node>
</launch>


<launch>
  <master auto="start"/>
  <param name="/use_sim_time" value="true"/>
  <include file="$(find navigation_stage)/move_base_config/move_base.xml"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(find navigation_stage)/stage_config/maps/mymap.pgm 0.05" respawn="false" />
  <!-- <node pkg="stage_ros" type="stageros" name="stageros" args="$(find navigation_stage)/stage_config/worlds/willow-pr2-5cm.world" respawn="false" >
    <param name="base_watchdog_timeout" value="0.2"/>
  </node> -->
  <include file="$(find navigation_stage)/move_base_config/amcl_node.xml"/>  
  <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find navigation_stage)/single_robot.rviz" /> -->
</launch>
